src/Plugin.php

```diff
@@ -0,0 +1,51 @@
+<?php
+namespace Hostnet\Component\Path;
+
+use Composer\Composer;
+use Composer\IO\IOInterface;
+use Composer\Plugin\PluginInterface;
+use Composer\Script\ScriptEvents;
+
+class Plugin implements PluginInterface
+{
+    /**
+     * @var Composer
+     */
+    private $composer;
+
+    /**
+     * @var IOInterface
+     */
+    private $io;
+
+    public static function getSubscribedEvents()
+    {
+        return [ScriptEvents::PRE_AUTOLOAD_DUMP => 'onPreAutoloadDump'];
+    }
+
+    public function activate(Composer $composer, IOInterface $io)
+    {
+        $this->composer = $composer;
+        $this->io       = $io;
+    }
+
+    public function onPreDumpAutoload()
+    {
+        $vendor_dir = realpath($this->composer->getConfig()->get('vendor-dir'));
+        $base_dir   = getcwd();
+        file_put_contents(
+            __DIR__ . '/Path.php',
+            <<<EOF
+<?php
+namespace Hostnet\Component\Path;
+
+class Path
+{
+    const VENDOR_DIR = '$vendor_dir';
+    const BASE_DIR   = '$base_dir';
```

Comment:
will do, for the few that really insist on having single quoted directory names...